### PR TITLE
React router 1.0

### PR DESCRIPTION
**Pull request by Pomax**
this PR updates the codebase to use react-router 1.0 -- `npm start` works, but a lot of the tests still fail, and there are a few API deprecation warnings, so there's a little bit of work left to be done, but slotting in React-Router 1.0 this time round was actually entirely doable, as opposed to absolutely impossible.

Do not merge yet (although if you want to test the site via `npm start`, by all means, do)

### CODE REVIEW

File: test/app.test.js [hunk 1]
```diff
@@ -72,14 +77,6 @@ describe('app', function() {
       .end(done);
   });
 
-  it('adds trailing slash to public HTML pages', function(done) {
```
Comment (mmmavis) [inline test/app.test.js]: should we keep this?

Comment (Pomax) [inline test/app.test.js]: the code no longer does this redirect, so we don't have to test for it.

File: components/page.jsx [hunk 2]
```diff
@@ -125,7 +135,13 @@ var Page = React.createClass({
           <div className="row">
             <Sidebar/>
             <main className="content col-md-9" role="main" id="content" tabIndex="-1">
-              <RouteHandler showModal={this.showModal} hideModal={this.hideModal}/>
+            {
+              React.cloneElement(this.props.children, {
```
Comment (mmmavis) [inline components/page.jsx]: could you explain why we are doing `cloneElement` vs just passing extra `props` to the existing element?

**Comment by mmmavis** [comment]
:clock5: = :chocolate_bar: :cake: :beers: :cocktail: :tada:

**Pull request merged**